record `kotlin.ResultKt` @kind class @AccessFlag PUBLIC_FINAL {
}
procedure `java.lang.Object` `createFailure`(`java.lang.Throwable` v1 @kind object) @signature `Lkotlin/ResultKt;.createFailure:(Ljava/lang/Throwable;)Ljava/lang/Object;` @AccessFlag PUBLIC_STATIC_FINAL {
    temp;
    v0;

  #L52ef28.  v0:= "exception" @kind object;
  #L52ef2c.  call `checkNotNullParameter`(v1, v0) @signature `Lkotlin/jvm/internal/Intrinsics;.checkNotNullParameter:(Ljava/lang/Object;Ljava/lang/String;)V` @kind static;
  #L52ef32.  v0:= new `kotlin.Result$Failure`;
  #L52ef36.  call `<init>`(v0, v1) @signature `Lkotlin/Result$Failure;.<init>:(Ljava/lang/Throwable;)V` @kind direct;
  #L52ef3c.  return v0 @kind object;
}
procedure `void` `throwOnFailure`(`java.lang.Object` v1 @kind object) @signature `Lkotlin/ResultKt;.throwOnFailure:(Ljava/lang/Object;)V` @AccessFlag PUBLIC_STATIC_FINAL {
    temp;
    v0;

  #L52ef50.  v0:= instanceof @variable v1 @type ^`kotlin.Result$Failure` @kind boolean;
  #L52ef54.  if v0 != 0 then goto L52ef5a;
  #L52ef58.  return @kind void;
  #L52ef5a.  v1:= (`kotlin.Result$Failure`) v1 @kind object;
  #L52ef5e.  v1:= v1.`kotlin.Result$Failure.exception` @type ^`java.lang.Throwable` @kind object;
  #L52ef62.  throw v1;
}
